src/metabase/sync/field_values.clj

```diff
@@ -45,23 +51,58 @@
                              (clear-field-values-for-field! field)))]
               (update-field-value-stats-count fv-change-counts result)))
           {:errors 0, :created 0, :updated 0, :deleted 0}
-          (db/select Field :table_id (u/the-id table), :active true, :visibility_type "normal")))
+          (table->fields-to-scan table)))
 
 (s/defn ^:private update-field-values-for-database!
-  [database :- i/DatabaseInstance]
-  (apply merge-with + (map update-field-values-for-table! (sync-util/db->sync-tables database))))
+  [_database :- i/DatabaseInstance
+   tables :- [i/TableInstance]]
+  (apply merge-with + (map update-field-values-for-table! tables)))
 
 (defn- update-field-values-summary [{:keys [created updated deleted errors]}]
   (trs "Updated {0} field value sets, created {1}, deleted {2} with {3} errors"
        updated created deleted errors))
 
-(def ^:private field-values-steps
-  [(sync-util/create-sync-step "update-field-values" update-field-values-for-database! update-field-values-summary)])
+(defn- delete-expired-advanced-field-values-summary [{:keys [deleted]}]
+  (trs "Deleted {0} expired advanced fieldvalues" deleted))
+
+(s/defn delete-expired-advanced-field-values-for-table!
+  "Delete all expired advanced FieldValues for a table.
+  For more info about advanced FieldValues, check the docs in [[metabase.models.field-values/field-values-types]]"
+  [table :- i/TableInstance]
+  (reduce (fn [acc field]
```

Comment:
This inner function could become its own function of just `field` that returns `row-count`, then this outer function becomes
```
(->> (table->fields-to-scan table)
     (map the-extracted-function)
     (reduce +))
```
which I think is clearer.

Reply:
it is!
Updated per your suggestion.